eth/filters/api.go

```diff
@@ -233,6 +232,38 @@ func (api *PublicFilterAPI) NewHeads(ctx context.Context) (*rpc.Subscription, er
 	return rpcSub, nil
 }
 
+// NewDeposits send a notification each time a new deposit received from bridge.
+func (api *PublicFilterAPI) NewDeposits(ctx context.Context, crit ethereum.FilterState) (*rpc.Subscription, error) {
+	notifier, supported := rpc.NotifierFromContext(ctx)
+	if !supported {
+		return &rpc.Subscription{}, rpc.ErrNotificationsUnsupported
+	}
+
+	rpcSub := notifier.CreateSubscription()
+	go func() {
+		stateData := make(chan *types.StateData)
+		stateDataSub := api.events.SubscribeNewDeposits(stateData)
+
+		for {
+			select {
+			case h := <-stateData:
+				if crit.Did == h.Did || crit.Contract == h.Contract ||
+					(crit.Did == 0 && crit.Contract == common.Address{}) {
```

Comment:
Please test this. This might not work. You have to check the bytes equality.
```
crit.Contract == common.Address{}
```

Reply:
If `Did` or `Contract` values is not set then is condition is working file
any specific case you want to refer here @jdkanani